pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -94,6 +95,15 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1.NodeClaim, no
 	stored := node.DeepCopy()
 	controllerutil.AddFinalizer(node, v1.TerminationFinalizer)
 
+	// Update cached allocatables
+	cacheMapKey := fmt.Sprintf(
+		"allocatableCache;%s;%s",
+		nodeClaim.Labels[v1.NodePoolLabelKey],
```

Comment:
What happens when `nodepool.spec.template.nodeClassRef` changes?

Reply:
Node hash recalculation triggers, and cache is flushed for that nodepool 

https://github.com/kubernetes-sigs/karpenter/pull/1420/files#diff-3fff53ac08298c9ad8e732e5fdeda2aea5018bc888160d6fa5fb432f4aad9f78R68

Reply:
Now that kubelet setting moved to nodeclass, this requires an update

Reply:
Based on [this](https://github.com/kubernetes-sigs/karpenter/blob/main/designs/drift-hash-versioning.md#summary-of-static-drift-hashing-mechanism), it seems that change in nodeclass spec is not supposed to trigger the hash recalculation, to avoid the unnecessary drift.

In this case, cache will be updated first time the node of this nodepool/instancetype comes up with updated allocatables